### DISCUSSION TITLE
Compare models before deploying

### DISCUSSION THREAD

**Discussion by craftanalytix** [TabularEditor/TabularEditor3#1297] [🙋♀️TE3 Q&A] [answered]
Is there anyway in TE3 to show the changes from Test to Prod before deploying to Prod?

Currently using ALM Toolkit that shows differences from both model before updating to target db

**Comment by mlonsk** [comment, marked as answer]
Hi @RoadDeal 

The primary way Tabular Editor currently supports is to source control your model and serialize in JSON or TMLD (Preview), then when you make changes you know which items change against your main branch, can assign review of the changes and can through GIT compare what is in your release branch vs. what is in your main/master branch. 

You can also open ALM toolkit using a C# script as described by another user here: https://github.com/TabularEditor/TabularEditor3/discussions/1088#discussioncomment-7264131
which could be a workaround to speed up the process if you wish to continue using ALM Toolkit.

**Reply by craftanalytix** [reply]
Thanks
